app/src/main/java/net/gini/android/gvlexample/GVLExampleActivity.java

```diff
@@ -0,0 +1,359 @@
+                .setMessage("Requirements not fulfilled:\n" + stringBuilder)
+                .setPositiveButton("OK", null)
+                .show();
+    }
+
+    @Override
+    public void requestCameraPermission(final PermissionRequestListener listener) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            listener.permissionGranted();
+            return;
+        }
+        Dexter.withActivity(this)
+              .withPermission(Manifest.permission.CAMERA)
+              .withListener(new PermissionListener() {
+                  @Override
+                  public void onPermissionGranted(final PermissionGrantedResponse response) {
+                      listener.permissionGranted();
+                  }
+
+                  @Override
+                  public void onPermissionDenied(final PermissionDeniedResponse response) {
+                      showCameraPermissionDeniedDialog(listener);
+                  }
+
+                  @Override
+                  public void onPermissionRationaleShouldBeShown(final PermissionRequest permission,
+                          final PermissionToken token) {
+                      showCameraPermissionRationale(token);
+                  }
+              })
+              .withErrorListener(new PermissionRequestErrorListener() {
+                  @Override
+                  public void onError(final DexterError error) {
+                      Log.e("dexter", error.name());
+                  }
+              })
+              .check();
+    }
+
+    @SuppressLint("InlinedApi")
+    @Override
+    public void requestStoragePermission(final PermissionRequestListener listener) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            listener.permissionGranted();
+            return;
+        }
+        Dexter.withActivity(this)
+              .withPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
+              .withListener(new PermissionListener() {
+                  @Override
+                  public void onPermissionGranted(final PermissionGrantedResponse response) {
+                      listener.permissionGranted();
+                  }
+
+                  @Override
+                  public void onPermissionDenied(final PermissionDeniedResponse response) {
+                      showStoragePermissionDeniedDialog(listener);
+                  }
+
+                  @Override
+                  public void onPermissionRationaleShouldBeShown(final PermissionRequest permission,
+                          final PermissionToken token) {
+                      showStoragePermissionRationale(token);
+                  }
+              })
+              .check();
+    }
+
+    private void showStoragePermissionDeniedDialog(final PermissionRequestListener listener) {
+        final AlertDialog alertDialog = new AlertDialog.Builder(this)
+                .setMessage("Stored images and PDFs can be analyzed only with storage access.")
```

Comment:
I guess that these hardcoded strings would change in a future PR, but maybe you can add them to your _strings_ file.